File: gruppe_e_main/frontend/src/app/models/RideMatchVO.ts


import { LatLng } from "./LatLng";
import { PaymentStatus } from "./PaymentStatus";
import { SimulationStatus } from "./SimulationStatus";

//Joseph
export interface User {
  id: number;

}
export interface RideMatchVO {
    id: number;
    startPoint: LatLng;
    stopovers?: LatLng[],
    endPoint: LatLng;
    customer?: User; //Joseph
    driver?: User;//Joseph
    totalDistanceKm: number;
    totalDurationMin: number;
    expectedPrice: number;
    simulationStatus: SimulationStatus;
    paymentStatus: PaymentStatus;
    ratingCustomer: number | null;
    ratingDriver: number | null
}

File: gruppe_e_main/backend/src/main/java/com/sepdrive/model/RideMatchVO.java
package com.sepdrive.model;

import lombok.Data;

import java.util.List;

@Data
public class RideMatchVO {

    private Long id;
    private LatLng startPoint;
    private List<LatLng> stopovers;
    private LatLng endPoint;
    private double totalDistanceKm;
    private double totalDurationMin;
    private double expectedPrice;
    private SimulationStatus simulationStatus;
    private Integer currentSimulationIndex;
    private PaymentStatus paymentStatus;
    private Integer ratingCustomer;
    private Integer ratingDriver;

    // joseph
    private UserVO customer;
    private UserVO driver;
}
